src/basic/user-util.c

```diff
@@ -602,3 +604,27 @@ bool valid_home(const char *p) {
 
         return true;
 }
+
+int maybe_setgroups(size_t size, const gid_t *list) {
+        static int cached_can_setgroups = -1;
```

Comment:
that won't change the /proc/ID/setgroups value for the new process.  It could be a problem if we were caching "allow" then cloning a new user namespace and change for the new userspace setgroups to "deny" before set its gid_map file.